packages/@lwc/engine-server/src/__tests__/fixtures/barrel-export-lwc-all/modules/x/cmp/cmp.js

```diff
@@ -0,0 +1,31 @@
+// Everything else can be imported in a helper, so we must check in the helper
+import {
+    LightningElement,
+    getComponentDef,
+    isComponentConstructor,
+    createContextProvider,
+    readonly,
+    setFeatureFlagForTest,
+    unwrap,
+    createElement,
+    renderComponent,
+} from '../../../barrel.js';
+
+// "Using" the imports so they don't get removed by the compiler
+console.log(
+    LightningElement,
+    // The LWC compiler doesn't let us use decorators like this, so we don't need to check them here
+    // api,
+    // track,
+    // wire,
+    getComponentDef,
+    isComponentConstructor,
+    createContextProvider,
+    readonly,
+    setFeatureFlagForTest,
+    unwrap,
+    createElement,
+    renderComponent
+);
```

Comment:
I'd still love for us to do _something_ with these rather than just logging them. E.g. exposing it on the component so that the template can prove that it rendered something, and that the exports we got from here are the same ones exported from `@lwc/ssr-runtime` (i.e. no duplicate modules).

Also, logging here will make our tests more noisy. We already have noisy logs, but this just adds to the problem. We could do something like `Promise.resolve(...)` to avoid the compiler removing it.

Reply:
>  I'd still love for us to do something with these rather than just logging them.  E.g. exposing it on the component so that the template can prove that it rendered something

It's a little crude, but what do you think about exposing them in the template via `inner-html`? Example [here](https://github.com/salesforce/lwc/pull/5034/files#diff-797e54a12cdf026edac4244c9fde36b7d1fc41428d219111344210e6171d2632R2). The function implementations would differ between v1 and v2 ofc so I trimmed them... E.g. `getComponentDef` in [V1](https://github.com/salesforce/lwc/blob/5d01843a7733a03b9ccb59a70ad64af955f15b88/packages/%40lwc/engine-core/src/framework/def.ts#L370) vs [V2] (stub) (https://github.com/salesforce/lwc/blob/5d01843a7733a03b9ccb59a70ad64af955f15b88/packages/%40lwc/ssr-runtime/src/stubs.ts#L19)

> Also, logging here will make our tests more noisy. We already have noisy logs, but this just adds to the problem.

Ok makes sense. See above.